sql/scan.go

```diff
@@ -350,6 +353,12 @@ func (n *scanNode) initTargets(targets parser.SelectExprs) error {
 			return n.err
 		}
 	}
+	for _, render := range n.render {
+		if _, n.err = parser.TypeCheckExpr(render); n.err != nil {
```

Comment:
Any reason not to call `TypeCheckExpr` before calling `addRender`? Or perhaps, call `TypeCheckExpr` in `addRender`.